profile/src/main/java/com/scottlogic/deg/profile/reader/CoreAtomicTypesConstraintReaderSource.java

```diff
@@ -302,7 +306,40 @@ else if (stringValidatedValue.isPresent()) {
                         ),
                         rules
                     )
+            ),
+            new ConstraintReaderMapEntry(
+                AtomicConstraintType.IS_FROM_FILE.getText(),
+                ".*",
+                (dto, fields, rules) -> {
+                    String value = ConstraintReaderHelpers.getValidatedValue(dto, String.class);
+                    String withEnding = appendCsvIfNotPresent(value);
+
+                    PathToStringsLoader loader = new CSVFromPathToStringsLoader();
+                    Stream<String> names = loader.retrieveNames(withEnding);
+
+                    Set<Object> strings = names
+                        .map(ConstraintReaderHelpers::downcastToObject)
+                        .collect(Collectors.toSet());
+
+                    Field field = fields.getByName(dto.field);
+                    return new AndConstraint(
+                        new IsInSetConstraint(field, strings, rules),
+                        new IsOfTypeConstraint(field, IsOfTypeConstraint.Types.STRING, rules)
+                    );
+                }
             )
         );
     }
+
+    private String appendCsvIfNotPresent(final String path) {
```

Comment:
I'd prefer if we didn't do this.
with "fromFile" it would make sense for the profile to require a file extension.